python-threatexchange/threatexchange/cli/match_cmd.py

```diff
@@ -145,104 +146,41 @@ def __init__(
-            return
-
         for path in self.files:
-            for s_type, index in indices:
-                seen = set()  # TODO - maybe take the highest certainty?
-                if self.as_hashes:
-                    results = _match_hashes(path, s_type, index)
-                else:
-                    results = _match_file(path, s_type, index)
-
-                for r in results:
-                    metadatas: t.List[t.Tuple[str, FetchedSignalMetadata]] = r.metadata
-                    for collab, fetched_data in metadatas:
-                        if not self.all and collab in seen:
-                            continue
-                        seen.add(collab)
-                        # Supposed to be without whitespace, but let's make sure
-                        distance_str = "".join(r.similarity_info.pretty_str().split())
-                        print(
-                            s_type.get_name(),
-                            distance_str,
-                            f"({collab})",
-                            fetched_data,
-                        )
-
-
-def _match_file(
-    path: pathlib.Path, s_type: t.Type[SignalType], index: SignalTypeIndex
-) -> t.Sequence[IndexMatch]:
-    if issubclass(s_type, MatchesStr):
-        return index.query(path.read_text())
-    assert issubclass(s_type, FileHasher)
-    return index.query(s_type.hash_from_file(path))
-
-
-def _match_hashes(
-    path: pathlib.Path, s_type: t.Type[SignalType], index: SignalTypeIndex
-) -> t.Sequence[IndexMatch]:
-    ret: t.List[IndexMatch] = []
-    for hash in path.read_text().splitlines():
-        hash = hash.strip()
-        if not hash:
-            continue
-        try:
-            hash = s_type.validate_signal_str(hash)
-        except Exception:
-            logging.exception("%s failed verification on %s", s_type.get_name(), hash)
-            hash_repr = repr(hash)
-            if len(hash_repr) > 50:
-                hash_repr = hash_repr[:47] + "..."
-            raise CommandError(
-                f"{hash_repr} from {path} is not a valid hash for {s_type.get_name()}",
-                2,
-            )
-        ret.extend(index.query(hash))
-    return ret
+            seen = set()
+            if self.as_hashes:
+                results = self.matcher.match(*path.read_text().splitlines())
+            else:
+                results = self.matcher.match(path)
+            for r in results:
+                metadatas: t.List[t.Tuple[str, FetchedSignalMetadata]] = r.metadata
+                for collab, fetched_data in metadatas:
+                    if not self.all and collab in seen:
+                        continue
+                    seen.add(collab)
+                    # Supposed to be without whitespace, but let's make sure
+                    distance_str = "".join(r.similarity_info.pretty_str().split())
+                    print(
+                        # s_type.get_name(),
```

Comment:
hmm, knowing the type matched is valuable. One options though I'm not sure it is worth the overhead is to have it as part of the metadata(s)

Reply:
Yeah, agreed, and something that we/I need to come back to. I commented it out for now as it didn't exist in this scope, and didn't want to forget that I removed it.